crates/core/src/operations/cast.rs

```diff
@@ -1,36 +1,106 @@
 //! Provide common cast functionality for callers
 //!
-use arrow_array::{Array, ArrayRef, RecordBatch, StructArray};
+use arrow::datatypes::DataType::Dictionary;
+use arrow_array::{new_null_array, Array, ArrayRef, RecordBatch, StructArray};
 use arrow_cast::{cast_with_options, CastOptions};
-use arrow_schema::{DataType, Fields, SchemaRef as ArrowSchemaRef};
-
+use arrow_schema::{
+    ArrowError, DataType, Field as ArrowField, Fields, Schema as ArrowSchema,
+    SchemaRef as ArrowSchemaRef,
+};
 use std::sync::Arc;
 
 use crate::DeltaResult;
 
+pub(crate) fn merge_field(left: &ArrowField, right: &ArrowField) -> Result<ArrowField, ArrowError> {
```

Comment:
Why did you opt for creating your own function instead of the try_merge in arrow directly?

Reply:
Because of the Dictionary Type, which is how I get partitioned values. I need to be able to allow DictionaryType<_, value_type> == value_type